### PR TITLE
Ft/admin login endpoint

### PR DESCRIPTION
**Pull request by OnyekachiSamuel**
#### What does this PR achieve?
The implementation for the admin login endpoint was achieved in this PR
#### Any background context?
None
#### Manual Testing Steps
You can test this out in Postman. First, create a hash for the admin password you would like to use, this you will use in seeding the database for the admin user. Then, try to login to the endpoint /admin/login with the admin email and password.
#### Relevant Trello Tickets
- [Admin Login Ticket](https://trello.com/c/1UmglWd3/9-admin-login)

### CODE REVIEW

File: api/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    adminLogin: (req, res) => {
+        const { email, password } = req.body
+        const errors = validationResult(req);
+
+        if (!errors.isEmpty()) {
+          return res.status(422).json({ errors: errors.array() });
+        }
+
+        models.user.findOne({where: { email }}).then((user) => {
+            if (user && user.dataValues.email === email && user.dataValues.isAdmin === true) {
```
Comment (chidioguejiofor) [inline api/controllers/user.js]: I think the test for email here is a bit redundant. I suggest we use:

```suggestion
            if (user &&  user.dataValues.isAdmin === true) {
            
```

File: api/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                const check = bcrypt.compareSync(password, user.dataValues.password);
+                const payload = {
+                    email: user.email,
+                    password: user.password
+                }
+                if (check) {
+                    res.status(200).json({data: payload})
```
Comment (chidioguejiofor) [inline api/controllers/user.js]: Do you think it is best to throw in JWT based auth here or should it be in another commit?

**Review by chidioguejiofor — COMMENTED** [review]
I think this is cool just a few comments. I think this would be good to go once you add some `Behavioural` tests

File: api/controllers/user.js (hunk 1 above)
Comment (OnyekachiSamuel) [reply api/controllers/user.js]: Okay, noted, thanks.

File: api/controllers/user.js (hunk 2 above)
Comment (OnyekachiSamuel) [reply api/controllers/user.js]: You mean generating a token and send it as part of the response when a user logs in ? @chidioguejiofor

Comment (chidioguejiofor) [reply api/controllers/user.js]: Yeah that's what I meant. I think `isAdmin` and `email` should be enough.

File: api/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        models.user.findOne({where: { email }}).then((user) => {
+            if (user && user.dataValues.isAdmin === true) {
+                const check = bcrypt.compareSync(password, user.dataValues.password);
+                const payload = {
+                    email: user.email,
+                    password: user.password
+                }
+                if (check) {
+                    const token = jwt.sign({ email: user.email, isAdmin: user.dataValues.isAdmin }, process.env.SECRET_KEY, {
+                        expiresIn: 60 * 60 * 24
```
Comment (chidioguejiofor) [inline api/controllers/user.js:54]: Nice

File: api/controllers/user.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                      });
+                    res.status(200).json({data: payload, token})
```
Comment (chidioguejiofor) [inline api/controllers/user.js:56]: I think you should not return the password in the response. I think we can just return `name, email and isAdmin`

 Something like:
```suggestion
                    res.status(200).json({data: {
                       email: user.email,
                       name: user.name,
                       isAdmin: user.isAdmin,
                    }, token})
```

Is okay

**Review by chidioguejiofor — COMMENTED** [review]
Looks good. Just the password in the response is what needs to be fixed.

File: api/controllers/user.js (hunk 4 above)
Comment (OnyekachiSamuel) [reply api/controllers/user.js:56]: Yea, it doesn't make sense returning password, I don't think it will be needed, security-wise its not good at all, I just followed what we had in the trello board.

Comment (chidioguejiofor) [reply api/controllers/user.js:56]: Sorry about the poor description on trello

Comment (chidioguejiofor) [reply api/controllers/user.js:56]: Would try to make it better to subsequent tasks

**Comment by chidioguejiofor** [comment]
This looks good. Just squash and merge this on in
